samples/boards/esp32/light_sleep/src/main.c

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2022 Espressif Systems (Shanghai) Co., Ltd.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <zephyr/kernel.h>
+#include <zephyr/pm/pm.h>
+#include <zephyr/pm/device.h>
+#include <zephyr/pm/policy.h>
+#include <zephyr/drivers/gpio.h>
+#include "esp_sleep.h"
+
+/* Most development boards have "boot" button attached to GPIO0.
+ * You can also change this to another pin.
+ */
+#define SW0_NODE	DT_ALIAS(sw0)
+
+#if !DT_NODE_HAS_STATUS(SW0_NODE, okay)
+#error "unsupported board: sw0 devicetree alias is not defined"
+#endif
+
+/* Add an extra delay when sleeping to make sure that light sleep
+ * is chosen.
+ */
+#define LIGHT_SLP_EXTRA_DELAY	(50UL)
+
+static const struct gpio_dt_spec button =
+	GPIO_DT_SPEC_GET_OR(SW0_NODE, gpios, {0});
+
+static const struct pm_state_info pwr_state_info[] =
+	PM_STATE_INFO_LIST_FROM_DT_CPU(DT_NODELABEL(cpu0));
```

Comment:
this is retrieving all states info just to obtain a residency time. You can just obtain that manually.